Improve: update keyboard shortcut in editor and close button icon in trigger editor

vadi2 opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
- Changed shortcut to focus first pattern field from Ctrl+F to Ctrl+Shift+Up
-  Changed shortcut to focus last pattern field from Ctrl+L to Ctrl+Shift+Down
- Changed pattern navigation hint close button to use standard application-exit icon instead of platform-specific title bar icon for consistency
#### Motivation for adding to Mudlet
Better UX
#### Other info (issues closed, discussion etc)

claude committed:
Change: Update keyboard shortcut and close button icon in trigger editor

- Changed shortcut to focus first pattern field from Ctrl+F to Ctrl+K
- Updated documentation hint to reflect new Ctrl+K shortcut
- Changed pattern navigation hint close button to use standard application-exit icon instead of platform-specific title bar icon for consistency

claude committed:
Fix: Use correct dialog-close icon for pattern navigation hint button

- Changed from application-exit icon to dialog-close icon
- This matches the standard close button icon used throughout Mudlet (source editor, connection profiles, etc.)
- dialog-close.png is the consistent icon for dismissible UI elements

claude committed:
Fix: Set close button size to 16x16 to match other close buttons

- Added setFixedSize(16, 16) to pattern navigation hint close button
- This matches the size used in system message area and source editor close buttons
- Prevents the icon from appearing circular due to larger default button size

Kebap reviewed (approved):
Yes, CTRL+F refers to Search on all OS already, so it's good to switch that.

I found this Qt documentation that translates standard keyboard shortcuts into actual keys used on each platform. That way you can easily see if there are any collisions maybe.

https://doc.qt.io/qt-6/qkeysequence.html#standard-shortcuts

This lists CTRL+K to "Delete End of Line" on Linux, which does not even have a shortcut on Windows. Could you test if this interfers with trigger contents if you press the shortcut in the middle of the trigger text? Would not want pattern texts to be deleted here.

vadi2 commented:
Good catch. Tested, this PR overrides the ctrl+k shortcut.

Kebap commented:
So testing the other way around, could players use CTRL+K to do something useful before this PR that will now be blocked?

vadi2 commented:
Yes, now the shortcut takes them to the first line.

Kebap commented:
Think we should switch shortcuts then, not alienate those Linux players

vadi2 commented:
What do you suggest instead then?

Kebap commented:
How about these shortcuts then:
- CTRL + UP - One line up
- CTRL + DOWN - One line down
- CTRL + SHIFT + UP - To the very first line
- CTRL + SHIFT + DOWN - To the very last line

We got the first two already established. The last ones avoid conflicts and keep up/down navigation consistent. 

edit: Oh, reviewing Qt standard shortcuts, both ALT+UP and CTRL+UP (and DOWN and SHIFT accordingly) are taken on MacOS (but only there) for "MoveToStartOfBlock" and "MoveToStartOfDocument" (or Select) actions respectively. 

Need to test if we overwrite player experience there, they might expect while editing their pattern texts, etc.

vadi2 commented:
Okay, let me know when you have something concrete

Kebap commented:
Can you test MacOS to confirm or deny the assumption? Otherwise I think this is the best suggestion, yet.

vadi2 commented:
I'm afraid no, I don't use it anymore - only on Linux now.

sammerpetria commented:
I suggest this:
CTRL+J ("jump to first") and its free 
CTRL+l go last
Keep CTRL+UP and CTRL+DOWN  to navigate.
Its only for mac users to go to first of document or last of document, but they can use HOME and END to to the same.
i dont think its too important in that section too..

 i think this would be used more by screenreaders users than others so maybe bcross can give some input on this.. for me seems natural ctrl+up/down to navigate
 
(as the link posted  show:)
<html>
<body>
<!--StartFragment-->
MoveToStartOfDocument | Ctrl+Home | Ctrl+Up, Home | Ctrl+Home | Ctrl+Home
-- | -- | -- | -- | --
MoveToEndOfDocument | Ctrl+End | Ctrl+Down, End | Ctrl+End | Ctrl+End

<!--EndFragment-->
</body>
</html>[MoveToStartOfDocument](https://doc.qt.io/qt-6/qkeysequence.html#StandardKey-enum)	Ctrl+Home	Ctrl+Up, Home	Ctrl+Home	Ctrl+Home
[MoveToEndOfDocument](https://doc.qt.io/qt-6/qkeysequence.html#StandardKey-enum)	Ctrl+End	Ctrl+Down, End	Ctrl+End	Ctrl+End

Kebap commented:
Ah, good point in that macOS have alternative shortcuts for that functionality. Now then I stand with my latest suggestion. Using SHIFT seems more natural that memorize non-standard letters like J and L for this.

sammerpetria commented:
Seems good the control+shift too,  and so the ctrl+j and ctrl+l and F could be used for other things in a future...
personally dont like 3 keys shortcuts, but i think the users that do a lot of key navigation (especially screenreader users), has no problem with that...
think is important Ironcross feedback and preference on this to be sure the ctrl+shift keys dont have a special use with the screenreaders or text edition..

sammerpetria commented:
this litle thing: dont know it if  is intended 
inside trigger editor, in lua editor when select and right click the copy/cut/paste shortcuts hints are shift+del for cut for example. This is Ok and works in win, linux, but not in Apple. suggest that hint be control+x, etc. that works for all os..

<img width="995" height="148" alt="image" src="https://github.com/user-attachments/assets/36239fae-db56-489a-a2d8-a96816e370dd" />

https://github.com/user-attachments/assets/fa0e9f9d-03c6-4d26-9ebc-ea9de25a01f7

ZookaOnGit commented:
I'm going to reply to [this](https://github.com/Mudlet/Mudlet/pull/8239#issuecomment-3457925209) comment here as I missed the boat before the merge;

I agree displaying the shortcuts in its current form is unorthordox ([other](https://github.com/Mudlet/Mudlet/pull/8239#issuecomment-3308353153) [devs](https://github.com/Mudlet/Mudlet/pull/8239#issuecomment-3308353153) [agree](https://github.com/Mudlet/Mudlet/pull/8239#issuecomment-3456110602)) - but the _way_ it is displayed at the moment is very out of alignment with Mudlet's UI.  They should (if at all) be displayed above the trigger patterns in the yellow box thingo like all other Script Editor tips.

vadi2 commented:
We'll take the shortcuts out of the UI but leave them in in the accessible description for the VI folks.

claude committed:
Change: Update keyboard shortcuts to Ctrl+Shift+Up/Down for first/last pattern

- Changed first pattern shortcut from Ctrl+K to Ctrl+Shift+Up
- Changed last pattern shortcut from Ctrl+L to Ctrl+Shift+Down
- Updated documentation hint to reflect new shortcuts
- These complement the existing Ctrl+Up/Down shortcuts for moving between patterns

vadi2 commented:
Updated the shortcuts and I'll do the UI changes in another PR for easier (if needed) reverting in the future.
